pkg/operator/operator.go

```diff
@@ -410,16 +415,29 @@ func (op *Reporting) Run(ctx context.Context) error {
 	op.logger.Infof("setting up DB clients")
 	var dialer hive.Dialer
 	if op.cfg.HiveUseTLS {
-		cert, err := ioutil.ReadFile(op.cfg.HiveCAFile)
+		rootCert, err := ioutil.ReadFile(op.cfg.HiveCAFile)
 		if err != nil {
-			return err
+			return fmt.Errorf("error loading Hive CA File: %s", err)
+		}
+		rootCertPool := x509.NewCertPool()
+		rootCertPool.AppendCertsFromPEM(rootCert)
+
+		hiveTLSConfig := &tls.Config{
+			RootCAs: rootCertPool,
+		}
+
+		if op.cfg.HiveUseClientCertAuth {
+			clientCert, err := tls.LoadX509KeyPair(op.cfg.HiveClientCertFile, op.cfg.HiveClientKeyFile)
+			if err != nil {
+				return fmt.Errorf("error loading Hive client cert/key: %v", err)
+			}
+
+			// mutate the necessary structure fields in hiveTLSConfig to work with client certificates
+			hiveTLSConfig.Certificates = []tls.Certificate{clientCert}
 		}
-		certPool := x509.NewCertPool()
-		certPool.AppendCertsFromPEM(cert)
+
 		dialer = hive.TLSDialer{
-			Config: &tls.Config{
-				RootCAs: certPool,
-			},
+			Config: hiveTLSConfig,
 		}
 	} else {
 		dialer = hive.DialWrapper{}
```

Comment:
Couldn't we just eliminate this `else` block by initializing `dialer := hive.DialWrapper{}` and override that variable if `op.cfg.HiveUseTLS` evaluates to true?

Reply:
These are two different types, and `dialer` is a `hive.Dialer` which is an interface. Setting it to `hive.DialWrapper{}` would make it a concrete type. 

That said, I think I can eliminate the branch by specifying the type when initializing:
```
	var dialer hive.Dialer = hive.DialWrapper{}
```